ReactQt/runtime/src/netinfo.cpp

```diff
@@ -11,41 +11,111 @@
+    void monitorNetworkAccess();
+
+    void setNetworkState(NetworkState state);
 
     Bridge* bridge = nullptr;
+    QTimer requestTimer;
+    QTimer replyTimer;
+    NetworkState networkState = UnkownState;
+    int replyStateCheckCounter = 0;
+
+Q_SIGNALS:
+    void networkStateChanged();
 };
 
+namespace {
+static QMap<NetInfoPrivate::NetworkState, QString> accessibleName{{NetInfoPrivate::UnkownState, "unknown"},
+                                                                  {NetInfoPrivate::NotAvailable, "none"},
+                                                                  {NetInfoPrivate::Available, "wifi"}};
+} // namespace
+
+void NetInfoPrivate::setNetworkState(NetInfoPrivate::NetworkState state) {
+    if (networkState != state) {
+        networkState = state;
+        Q_EMIT networkStateChanged();
+    }
+}
+
+NetInfoPrivate::NetInfoPrivate() {
+    replyTimer.setInterval(NETWORK_REPLY_CHECK_TIMEOUT);
+}
+
+NetInfoPrivate::~NetInfoPrivate() {
+    requestTimer.stop();
+    replyTimer.stop();
+}
+
+void NetInfoPrivate::monitorNetworkAccess() {
+    QObject::connect(this, &NetInfoPrivate::networkStateChanged, [=]() {
+        bridge->eventDispatcher()->sendDeviceEvent("networkStatusDidChange",
+                                                   QVariantMap{{"connectionType", accessibleName.value(networkState)}});
+    });
+
+    QObject::connect(&requestTimer, &QTimer::timeout, [=]() {
+        QNetworkRequest req(QUrl("http://www.google.com"));
+
+        replyStateCheckCounter = 0;
+        replyTimer.start();
+
+        QNetworkReply* reply = bridge->networkAccessManager()->head(req);
+        auto replyFinishOrTimeout = [=]() {
+            auto replyFinishedNoError = reply->isFinished() ? reply->error() == QNetworkReply::NoError : false;
+            auto previousNetworkState = (networkState == NetInfoPrivate::Available);
+
+            if (++replyStateCheckCounter >= MAX_REPLY_CHECK_COUNTER || replyFinishedNoError) {
+                replyTimer.stop();
+
+                if (replyFinishedNoError != previousNetworkState) {
+                    auto newNetworkState =
+                        replyFinishedNoError ? NetInfoPrivate::Available : NetInfoPrivate::NotAvailable;
+                    qDebug() << "monitorNetworkAccess: setNetworkAccessible "
+                             << "old: " << previousNetworkState << "new: " << newNetworkState;
+                    setNetworkState(newNetworkState);
+                }
+                reply->deleteLater();
+            }
+        };
+
+        QObject::connect(&replyTimer, &QTimer::timeout, reply, replyFinishOrTimeout);
+        replyTimer.start();
+    });
+
+    requestTimer.start(NETWORK_REQUEST_TIMEOUT);
+}
 void NetInfo::getCurrentConnectivity(const ModuleInterface::ListArgumentBlock& resolve,
                                      const ModuleInterface::ListArgumentBlock& reject) {
     Q_UNUSED(reject);
     Q_D(NetInfo);
-    resolve(d->bridge,
```

Comment:
curios if that was removed, do we still need `NetInfo::getCurrentConnectivity` defined in general?